jax/experimental/mosaic/gpu/utils.py

```diff
@@ -1115,6 +1124,80 @@ def memref_ptr(memref_arg, memory_space=None):
   )
 
 
+def extern_elementwise_func_apply(func, return_type, arg_list,
+                                  static_arg_idxs=(), pack_elts=False,
+                                  elt_byte_count_idx=None):
+  tidx = thread_idx()
+  wgidx = warpgroup_idx()
```

Comment:
perhaps use `sync=False` here to avoid unnecessarily putting this in uniform registers